### PR TITLE
Bug 564328 API revision | conditions | rethink key interfaces

### PR DESCRIPTION
**Pull request by eparovyshnaya**
- 'LicensingCondition' interface reconstructed

Signed-off-by: elena.parovyshnaya <elena.parovyshnaya@gmail.com>

### CODE REVIEW

File: bundles/org.eclipse.passage.lic.api/src/org/eclipse/passage/lic/internal/api/conditions/EvaluationType.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		}
+		return identifier.equals(((EvaluationType) object).identifier);
+	}
+
+	@Override
+	public final String toString() {
+		return identifier;
+	}
+
+	public static final class Time extends EvaluationType {
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.lic.api/src/org/eclipse/passage/lic/internal/api/conditions/EvaluationType.java:57]: This is not yet implemented and _may be_ we can add it later.  Or just remove it to not overlap with `ValidityPeriod`
The initial idea was to support `cron`-like expressions.

File: bundles/org.eclipse.passage.lic.api/src/org/eclipse/passage/lic/internal/api/conditions/VersionMatch.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ * <p>
+ * This way or another a license defines which version of a feature is allowed
+ * to use. It also defines <i>how far</i> from this specified version can be the
+ * actual version of the feature that has been plugged to the product under
+ * licensing.
+ * </p>
+ * 
+ * @see Condition
+ */
+public interface VersionMatch {
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.lic.api/src/org/eclipse/passage/lic/internal/api/conditions/VersionMatch.java:30]: Need to revisit this to ensure rich version range support, most probably it is sufficient in this form

File: bundles/org.eclipse.passage.lic.api/src/org/eclipse/passage/lic/internal/api/conditions/EvaluationType.java (hunk 1 above)
Comment (eparovyshnaya) [reply bundles/org.eclipse.passage.lic.api/src/org/eclipse/passage/lic/internal/api/conditions/EvaluationType.java]: Removed

File: bundles/org.eclipse.passage.lic.api/src/org/eclipse/passage/lic/internal/api/conditions/VersionMatch.java (hunk 2 above)
Comment (eparovyshnaya) [reply bundles/org.eclipse.passage.lic.api/src/org/eclipse/passage/lic/internal/api/conditions/VersionMatch.java:30]: It's quite general:  
 - data part supplies any string, as compicated as you like, and the
 - algorithmic part `MatchingRule` knows how to deal with it. This one, again, is general. It demands nothing from data, it just matches two strings according it's hidden logic.

I suppose this is enough. Currently I'm reworking `LicensingVersions` on these interfaces - it goes well.

**Pull request merged**